### ISSUE TITLE
Implement the ability to query a single GA4 property by ID

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#3102]
Following up on #2979, it should also be possible to look up a single GA4 property by ID, in the `analytics-4` module class in PHP and the `modules/analytics-4` store in JS.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `Analytics_4` module class should have a new REST datapoint `GET:property`.
    * It should accept a `propertyID` request parameter, which is a plain property ID (without `properties/` prefix).
    * It should rely on the [properties.get](https://developers.google.com/analytics/devguides/config/admin/v1/rest/v1alpha/properties/get) API endpoint.
    * It should return the property object.
* The `modules/analytics-4` datastore should get a new selector `getProperty( propertyID )`.
    * It should rely on the `GET:modules/analytics-4/data/property` REST datapoint to get the property if it's not already in store state.
    * In other words, the existing `getProperties` resolver should also be updated to reduce the amount of necessary API calls: Any properties returned by `getProperties` should **also** be keyed by ID in state (instead of just under the account they belong to).

## Implementation Brief

* In the `includes/Modules/Analytics_4.php` file:
  * Add `'GET:property' => array( 'service' => 'analyticsadmin' )` to the returning array of the `get_datapoint_definitions` method.
  * Add a new case `GET:property` to the `create_data_request` method. It should return an error (like [here](https://github.com/google/site-kit-wp/blob/develop/includes/Modules/Analytics_4.php#L233-L240)) if `propertyID` is not set in the `$data` array. Otherwise it should return the result of the `$this->get_service( 'analyticsadmin' )->properties->get( $data['propertyID'] )` call.
* In the `modules/analytics-4/datastore/properties.js` file:
  * Create a new fetch store for the `property` endpoint:
    * Use `getProperty` base name.
    * Issue a GET request to the `modules/analytics-4/data/property` endpoint with `propertyID` query argument.
    * Set received property under appropriate `propertyID` key in the `propertiesByIDs`  property of the state object in the `reducerCallback` callback.
    * The `argsToParams` callback should convert the incoming `propertyID` argument to an object with `propertyID` property.
    * Make sure that `propertyID` is valid using the `isValidPropertyID` function in the `validateParams` callback.
  * Update the `reducerCallback` function of the `fetchGetPropertiesStore` datastore. In addition to the existing functionality, it should also populate the `propertiesByID` section of the state object using received properties.
  * Add a new selector `getProperty( propertyID )` to the `baseSelectors` object. It should return a property from the `propertiesByIDs` object using provided `propertyID` argument.
  * Add a new resolver `getProperty( propertyID )` to the `baseResolvers` object. It should call `getProperty.actions.fetchGetProperty( propertyID )` function if `getProperty( propertyID )` selector returns `undefined`.

### Test Coverage

* Add tests for the new `getProperty` selector.

### Visual Regression Changes

* N/A

## QA Brief

* Make sure that Analytics-4 module has `getProperty` selected and it works as mentioned in the AC section.
* Also check the `fetchGetPropertiesStore` datastore to verify that it adds received properties to the `propertiesByID` state as well.
* Finally, check the Analytics_4 module class and confirm that it has the new `GET:property` endpoint and it works as expected in the AC.

## Changelog entry

* Implement REST data point and JS data store selector to look up a single GA4 property by ID.

**Comment by felixarntz** [comment]
IB ✅

**Comment by felixarntz** [comment]
@eugene-manuilov Can you add a QA Brief to this one please?

**Comment by eugene-manuilov** [comment]
QAB is added.

**Comment by aaemnnosttv** [comment]
## QA ❌ 

* `getProperties` is not setting property IDs in `propertiesByID` properly. **Note the `undefined` key**  
![image](https://user-images.githubusercontent.com/1621608/115370671-f6748400-a1d1-11eb-9469-1dec3a8ce1ee.png)
* `getProperty` works as expected, but triggers an additional request when called after `getProperties`  
![image](https://user-images.githubusercontent.com/1621608/115370736-055b3680-a1d2-11eb-9fcd-1d9787bb9495.png)

**Comment by eugene-manuilov** [comment]
@aaemnnosttv yes, this is a known issue that is already fixed in the #3197. Once that PR is merged into develop, it will work correctly.

**Comment by felixarntz** [comment]
@eugene-manuilov Can you open a quick PR with only the relevant changes from #3197, targeted against this issue? While it doesn't really affect the release because the selector isn't used, we shouldn't technically ship an issue that's knowingly broken and only fix it later (#3104 will not go into the same release as it's too late for that).

**Comment by eugene-manuilov** [comment]
Added #3200

**Comment by ivankruchkoff** [comment]
@aaemnnosttv based on your previous QA, can you take a look? I wasn't able to figure out where to go to trigger the redux actions for the ga4setup.

**Comment by aaemnnosttv** [comment]
## QA ✅ 

- Selecting `getProperties( accountID )` adds properties to `state.propertiesByID` as well  
![image](https://user-images.githubusercontent.com/1621608/115669976-94db2380-a351-11eb-968d-c2f9702f7a74.png)
- Selecting `getProperty( propertyID )` right after returns the property right away  
![image](https://user-images.githubusercontent.com/1621608/115670130-bdfbb400-a351-11eb-9650-d7e3bf01dd21.png)
- Only a single request made to `google-site-kit/v1/modules/analytics-4/data/properties?accountID=`
